internal/api/v1/upload.go

```diff
@@ -27,68 +28,62 @@ func (hc ApplicationsController) Upload(w http.ResponseWriter, r *http.Request)
 
 	gitea, err := gitea.New()
 	if err != nil {
-		internalError(w, err.Error())
+		handleError(w, err, http.StatusInternalServerError)
 		return
 	}
 
 	log.V(2).Info("parsing multipart form")
 
 	file, _, err := r.FormFile("file")
 	if err != nil {
-		userError(w, "can't read multipart file input")
+		err = errors.Wrapf(err, "can't read multipart file input")
+		handleError(w, err, http.StatusBadRequest)
```

Comment:
Why removing user/internal error ? Could have been re-implemented using `handleError`. Less work, just two places to change.

Reply:
Maybe change the signature, to provide the `err` for wrapping.